### PR TITLE
Feature 17 page preview

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

File: src/components/layout-components/editor.tsx [hunk 1]
```diff
@@ -11,17 +22,79 @@ const EditorContainer = styled(Stack, {
...
+    ]
+  };
+
   return (
-    <EditorContainer direction="row" gap={4}>
+    <EditorContainer
+      direction="row"
+      gap={4}
+      onDoubleClick={ () => setPanelProperties(EditorPanelProperties.SURVEY) }
+    >
```
Comment (villejuutila) [inline src/components/layout-components/editor.tsx:81]: There seems to be a lot of inconsistence between code style. In some places we've got `prop={ value }` and some other places `prop={value}`. Wasn't Rome linter/formatter used in this? Rome autoformatter should handle all of that.

File: src/components/screens/preview-screen.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      <PreviewToolbar surveyName={ survey.title } />
+      <PreviewContainer>
+        <Preview
+          htmlString={ htmlTemplateDummy }
+          width={ DEVICE_WIDTH }
+          height={ DEVICE_HEIGHT }
+          scale={ EDITOR_SCREEN_PREVIEW_CONTAINER_WIDTH / DEVICE_WIDTH }
+        />
+      </PreviewContainer>
+      {/* TODO: the page count display */}
```
Comment (villejuutila) [inline src/components/screens/preview-screen.tsx:78]: Could this TODO be implemented?

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/components/layout-components/editor.tsx (hunk 1 above)
Comment (lauriLukkarinen) [reply src/components/layout-components/editor.tsx:156]: Currently we are following the industry standard, which is without gaps. The gaps should therefore only be present in older projects. Also, indeed, Rome linter has proven to be overall the best linter so far, so at least for now we should use it in new projects. It is easy to setup and it eliminates the debates over most style issues.

Comment (Jdallos) [reply src/components/layout-components/editor.tsx]: The linter has not being running as it should, I will create a seperate branch with the changes from the linter (as there will be quite a few).

File: src/components/pages/templates/template-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+  </head>
+  <body style="margin: 0;">
+    ${sanitizeHtml(bodyContent, sanitizeOptions)}
+  </body>
+  <script>
+    document.addEventListener("click", () =>
+      window.parent.postMessage("iFrameClick-${pageNumber}")
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/pages/templates/template-wrapper.ts:32]: Discussed and resolved

File: src/components/screens/preview-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+const PreviewScreen = () => {
+  const { id } = useParams();
+  const { surveysApi, layoutsApi, pagesApi } = useApi();
+  const setError = useSetAtom(errorAtom);
+  const { height } = useWindowSize();
+  const [ survey, setSurvey ] = useState<Survey>();
+  const [ surveyPages, setSurveyPages ] = useAtom(pagesAtom);
+  const [ pageLayouts, setPageLayouts ] = useAtom(layoutsAtom);
+  // TODO: update current page when changing pages in full screen preview.
+  const [ currentPage, _setCurrentPage ] = useState(1);
```
Comment (lauriLukkarinen) [inline src/components/screens/preview-screen.tsx:70]:
```suggestion
  const [ currentPage ] = useState(1);
```

File: src/components/screens/preview-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  /**
+   * Get the page layout based on page layout id
+   *
+   * @param page Page
+   * @returns layout html
+   */
+  const getPageLayout = (page: Page) => {
+    return pageLayouts.find(layout => layout.id === page.layoutId)?.html!;
```
Comment (lauriLukkarinen) [inline src/components/screens/preview-screen.tsx:148]:
```suggestion
    return pageLayouts.find(layout => layout.id === page.layoutId)!.html;
```

**Review by lauriLukkarinen — APPROVED** [review]

**Review by villejuutila — APPROVED** [review]

**Pull request merged**